apps/nowcasting-app/components/helpers/data.ts

```diff
@@ -0,0 +1,75 @@
+import { components } from "../../types/quartz-api";
+import { Map } from "../map";
+import { CombinedData, GspDeltaValue, MapFeatureObject } from "../types";
+import { FeatureCollection } from "geojson";
+import gspShapeData from "../../data/gsp_regions_20220314.json";
+import { formatISODateString, getRoundedPv, getRoundedPvPercent } from "./utils";
+import { get30MinNow } from "./globalState";
+import { SelectedData } from "../map/types";
+
+export const generateGeoJsonForecastData: (
+  forecastData?: components["schemas"]["OneDatetimeManyForecastValues"][],
+  targetTime?: string,
+  combinedData?: CombinedData,
+  gspDeltas?: Map<string, GspDeltaValue>
+) => { forecastGeoJson: FeatureCollection } = (
+  forecastData,
+  targetTime,
+  combinedData,
+  gspDeltas
+) => {
+  const gspForecastsDataByTimestamp = forecastData || [];
+  const gspShapeJson = gspShapeData as FeatureCollection;
+  const forecastGeoJson = {
+    ...gspShapeData,
+    type: "FeatureCollection" as "FeatureCollection",
+    features: gspShapeJson.features.map((featureObj, index) => {
+      const gspSystemInfo = combinedData?.allGspSystemData?.find(
+        (system) => system.gspId === index + 1
+      );
+      let selectedFC;
+      let selectedFCValue;
+      if (gspForecastsDataByTimestamp && targetTime) {
+        selectedFC = gspForecastsDataByTimestamp.find(
+          (fc) => fc.datetimeUtc.slice(0, 16) === formatISODateString(targetTime)
+        );
+        if (selectedFC) selectedFCValue = selectedFC.forecastValues[index + 1];
+      } else if (gspForecastsDataByTimestamp) {
+        let latestTimestamp = get30MinNow();
+        selectedFCValue = gspForecastsDataByTimestamp.find(
+          (fc) => fc.datetimeUtc.slice(0, 16) === latestTimestamp
+        )?.forecastValues[index];
+      }
+
+      const updatedFeatureObj: MapFeatureObject = {
+        ...featureObj,
+        properties: {
+          ...featureObj.properties,
+          [SelectedData.expectedPowerGenerationMegawatts]:
+            selectedFCValue && getRoundedPv(selectedFCValue),
+          [SelectedData.expectedPowerGenerationNormalized]:
+            selectedFCValue &&
+            // getRoundedPvPercent(selectedFCValue?.expectedPowerGenerationNormalized || 0),
```

Comment:
should this be commented out?

Reply:
getRoundedPvPercent(selectedFCValue / gspSystemInfo?.installedCapacityMw  || 0)

Reply:
@braddf this might need fixing before we merge

Reply:
Thanks @peterdudfield – this was the issue, I must have amended all the other parts of this call with the new object shape and forgotten to come back to this one!